packages/server/src/course/course.controller.ts

```diff
@@ -102,24 +103,69 @@ export class CourseController {
     // }
     const questions = new AsyncQuestionResponse();
     questions.helpedQuestions = all.filter(
-      question => question.status === asyncQuestionStatus.Resolved,
+      (question) => question.status === asyncQuestionStatus.Resolved,
     );
     questions.waitingQuestions = all.filter(
-      question => question.status === asyncQuestionStatus.Waiting,
+      (question) => question.status === asyncQuestionStatus.Waiting,
     );
     questions.otherQuestions = all.filter(
-      question =>
+      (question) =>
         question.status === asyncQuestionStatus.StudentDeleted ||
         question.status === asyncQuestionStatus.TADeleted,
     );
     questions.visibleQuestions = all.filter(
-      question =>
+      (question) =>
         question.visible === true &&
         question.status !== asyncQuestionStatus.TADeleted,
     );
     return questions;
   }
 
+  @Get('limited/:id/:code')
+  async getLimitedCourseResponse(
+    @Param('id') id: number,
+    @Param('code') code: string,
+  ): Promise<GetLimitedCourseResponse> {
+    const course = await CourseModel.findOne({
```

Comment:
Instead of making two separate requests to the Database, why don't we jam this in the OrganizationCourseModel. There is a relation between course model and organization course model tables, so I think you should be able to input where statement.

Reply:
But I need to return the course name and invite code. They are not present in the OrganizationCourseModel.

Reply:
you could use `relations` to fetch course stuff